### PR TITLE
🚧 Missing data work

### PR DESCRIPTION
**Pull request by veronikasamborska1994**

### CODE REVIEW

File: etl/steps/data/garden/missing_data/2024-03-14/children_out_of_school.meta.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+definitions:
+  common:
```
Comment (paarriagadap) [inline etl/steps/data/garden/missing_data/2024-03-14/children_out_of_school.meta.yml:2]: Maybe it's useful to add a `description_processing` field to explain how the data has major processing.

File: etl/steps/data/garden/missing_data/2024-03-14/children_out_of_school.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    ds_data = paths.load_dataset("gender_statistics")
+    tb = ds_data["gender_statistics"].reset_index()
+    tb_data = tb[["country", "year", "se_prm_uner"]]  # Interested columns.
+
+    ds_regions = paths.load_dataset("regions")
+    ds_income_groups = paths.load_dataset("income_groups")
+    ds_population = paths.load_dataset("population")
+    tb_population = ds_population["population"].reset_index()
+
+    # Clean data to include only years with data.
```
Comment (paarriagadap) [inline etl/steps/data/garden/missing_data/2024-03-14/children_out_of_school.py:43]: You can group these speficif steps in a function for more clarity in the main run function

File: etl/steps/data/garden/missing_data/2024-03-19/who_neuropsychiatric_conditions.meta.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+definitions:
+  common:
+    processing_level: major
```
Comment (paarriagadap) [inline etl/steps/data/garden/missing_data/2024-03-19/who_neuropsychiatric_conditions.meta.yml:3]: You can add `description_processing` to detail why processing is major

File: etl/steps/data/garden/missing_data/2024-03-19/who_neuropsychiatric_conditions.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    ds_data = paths.load_dataset("mortality_database")
+
+    tb = ds_data["neuropsychiatric_conditions__both_sexes__all_ages"].reset_index()
+    tb = add_origins_to_mortality_database(tb_who=tb)
+
+    tb_data = tb.filter(
+        items=["country", "year"]
+        + [col for col in tb if col.startswith("deaths_from_neuropsychiatric_conditions_per_100_000_people")]
+    )
+    print(tb_data)
```
Comment (paarriagadap) [inline etl/steps/data/garden/missing_data/2024-03-19/who_neuropsychiatric_conditions.py:44]: Delete print

File: etl/steps/data/garden/missing_data/2024-03-19/who_neuropsychiatric_conditions.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        tb_merged["deaths_from_neuropsychiatric_conditions_per_100_000_people_in__both_sexes_aged_all_ages"].isna(),
+        0,
+        1,
+    )
+
+    # Function to calculate missing data details.
+    def calculate_details(group_by_column):
+        # Group by year and group, then calculate total countries.
+        total_countries = tb_merged.groupby(["year", group_by_column]).size().reset_index(name="total_countries")
+        print(total_countries)
```
Comment (paarriagadap) [inline etl/steps/data/garden/missing_data/2024-03-19/who_neuropsychiatric_conditions.py:99]: Delete print if unneeded

File: etl/steps/data/garden/missing_data/2024-03-19/who_neuropsychiatric_conditions.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    all_combinations = pd.DataFrame(
+        [(country, year) for country in all_countries for year in all_years], columns=["country", "year"]
+    )
+
+    # Merge this DataFrame with tb_data
+    tb_data = pd.merge(all_combinations, tb_data, on=["country", "year"], how="left")
+
+    tb_data["region"] = tb_data["country"].map(country_to_region)
+    tb_data["income_group"] = tb_data["country"].map(country_to_income_group)
+    # print(tb_data)
```
Comment (paarriagadap) [inline etl/steps/data/garden/missing_data/2024-03-19/who_neuropsychiatric_conditions.py:83]: Delete comment

File: etl/steps/data/garden/missing_data/2024-03-19/who_neuropsychiatric_conditions.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
+"""Load a meadow dataset and create a garden dataset."""
```
Comment (paarriagadap) [inline etl/steps/data/garden/missing_data/2024-03-19/who_neuropsychiatric_conditions.py:1]: Same comments for children_out_of_school aply here. There might be a way to locate these function in a shared.py code so you can call it for both steps. It's not necessary this time though.

File: etl/steps/data/garden/missing_data/2024-03-25/who_md_suicides.meta.yml [hunk 8]
```diff
@@ -0,0 +1,49 @@
+definitions:
+  common:
+    processing_level: major
```
Comment (paarriagadap) [inline etl/steps/data/garden/missing_data/2024-03-25/who_md_suicides.meta.yml:3]: You can add `description_processing`

File: etl/steps/data/garden/missing_data/2024-03-25/who_md_suicides.py [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+    all_combinations = pd.DataFrame(
+        [(country, year) for country in all_countries for year in all_years], columns=["country", "year"]
+    )
+
+    # Merge this DataFrame with tb_data
+    tb_data = pd.merge(all_combinations, tb_data, on=["country", "year"], how="left")
+
+    tb_data["region"] = tb_data["country"].map(country_to_region)
+    tb_data["income_group"] = tb_data["country"].map(country_to_income_group)
+    # print(tb_data)
```
Comment (paarriagadap) [inline etl/steps/data/garden/missing_data/2024-03-25/who_md_suicides.py:83]: Remove comment

**Review by paarriagadap — APPROVED** [review]
Good! I added minor comments. I don't know if you should merge the Jupyter Notebook `exploration.ipynb`, since it's annoying to handle for changes on GitHub. One option is to use it like playground, in gitignore.

And run `make check` to delete unused imports :)

File: etl/steps/data/garden/missing_data/2024-03-14/children_out_of_school.meta.yml (hunk 1 above)
Comment (veronikasamborska1994) [reply etl/steps/data/garden/missing_data/2024-03-14/children_out_of_school.meta.yml]: thank you! goood point!

**Pull request merged**